chartsxblock/static/js/src/chartsxblock-studio.js

```diff
@@ -19,15 +25,71 @@ function ChartsXBlockStudio(runtime, element, data) {
                 }
             });
         } catch (e) {
-            console.log("Error: \n", e);
-            runtime.notify('cancel', {});
+            console.log("Error with stack trace: \n", e);
+            showError(runtime, e);
+        }
+    });
+
+    $( ".chart-bottom-add" ).click(function() {
+        chart.addRow();
+    });
+    $( ".chart-right-add" ).click(function() {
+        chart.addColumn();
+    });
+    $( ".chart-bottom-del" ).click(function() {
+        try{
+            chart.removeRow();
+        } catch (e) {
+            showError(runtime, e);
+        }
+    });
+    $( ".chart-right-del" ).click(function() {
+        try{
+            chart.removeColumn();
+        } catch (e) {
+            showError(runtime, e);
         }
     });
+
+}
+
+$.fn.makeEditable = function() {
+    $(this).on('dblclick',function(){
+        if($(this).find('input').is(':focus')) return this;
+        var cell = $(this);
+        var content = $(this).html();
+        $(this).html('<input type="text" value="' + $(this).html() + '" />')
+        .find('input')
+        .trigger('focus')
+        .on({
+        'blur': function(){
+                $(this).trigger('closeEditable');
+        },
+        'keyup':function(e){
+            if(e.which == '13'){ // enter
+                $(this).trigger('saveEditable');
+            } else if(e.which == '27'){ // escape
+                $(this).trigger('closeEditable');
+            }
+        },
+        'closeEditable':function(){
+            cell.html(content);
+        },
+        'saveEditable':function(){
+            content = $(this).val();
+            $(this).trigger('closeEditable');
```

Comment:
Moja greška, nisam vidio to. Probaj koristiti indentaciju malo više, čitljiviji će ti kod biti.